docs/src/main/asciidoc/opentelemetry.adoc

```diff
@@ -251,16 +251,90 @@ When creating manual instrumentation, while naming metrics or attributes you sho
 
-The `quarkus.otel.exporter.otlp.protocol` defaults to `grpc` but `http/protobuf` can also be used.
+CDI (Context Dependency Injection) will manage the exporters to use, according to the selected protocol or when applications implement their own CDI exporter, like in tests.
+
+The `quarkus.otel.exporter.otlp.protocol` property instructs Quarkus to switch the senders and defaults to `grpc` but `http/protobuf` can also be used.
 
 NOTE: If you change the protocol, you also need to change the port in the endpoint. The default port for `grpc` is `4317` and for `http/protobuf` is `4318`.
 
+=== Using CDI to produce a test exporter
+
+Leaving the default as CDI is particularly useful for tests. In the following example a Span exporter class is wired with CDI and then the telemetry can be used in test code.
+
+Creating a custom `SpanExporter` bean:
+
+[source,java]
+----
+    @ApplicationScoped
+    static class InMemorySpanExporterProducer {
+        @Produces
+        @Singleton
+        InMemorySpanExporter inMemorySpanExporter() {
+            return InMemorySpanExporter.create();
+        }
+    }
+----
+
+Where `InMemorySpanExporter` is a class from the OpenTelemetry test utilities dependency:
+
+[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
+.pom.xml
+----
+    <dependency>
+        <groupId>io.opentelemetry</groupId>
+        <artifactId>opentelemetry-sdk-testing</artifactId>
+        <scope>test</scope>
+    </dependency>
+----
+
+[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
+.build.gradle
+----
+implementation("io.opentelemetry:opentelemetry-sdk-testing")
+----
+
+The bean of that class can be injected to access the telemetry data. This is an example to obtain the spans:
+
+[source, java]
+----
+    @Inject
+    InMemorySpanExporter inMemorySpanExporter;
+
+    //...
+
+    List<SpanData> finishedSpanItems = inMemorySpanExporter.getFinishedSpanItems();
+----
+
+If this is used on an integration test project, you should access the class from inside the running process and not from the test class.
+A viable option could be to expose that data through a rest endpoint method:
+
+[source,java]
+----
+    @GET
+    @Path("/export")
+    public List<SpanData> exportTraces() {
+        return inMemorySpanExporter.getFinishedSpanItems()
+                .stream()
+                .filter(sd -> !sd.getName().contains("export")) <1>
+                .collect(Collectors.toList());
+    }
+----
+<1> This excludes calls to the export endpoint itself.
+
+For more details please take a look to the https://github.com/quarkusio/quarkus/blob/main/integration-tests/opentelemetry/src/main/java/io/quarkus/it/opentelemetry/ExporterResource.java[ExporterResource] in the Quarkus integration tests.
+
+=== The OpenTelemetry OTLP exporter
+
+This is currently not supported in quarkus. Configuration example for traces: `quarkus.otel.tracing.exporter=otlp`.
```

Comment:
```suggestion
This is currently not supported in Quarkus. Configuration example for traces: `quarkus.otel.tracing.exporter=otlp`.
```

I don't understand:

> Configuration example for traces

?

And I think I agree with @geoand that throwing a proper error in Quarkus for unsupported values would be better? Either with a fixed list of values if we know all of them or excluding the usual suspects people could be tempted to use?